feat(pairing): add config to let users enable message syncing

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/18894

Based on top of https://github.com/status-im/status-go/pull/6948
I will merge this one into the base once it's reviewed. It will enable the the tests to run and the coverage to go up on the base.

Adds a param for users to enable message syncing during local pairing.

Updates the test to use that param

Based on top of https://github.com/status-im/status-go/pull/6948
I will merge this one into the base once it's reviewed. It will enable the the tests to run and the coverage to go up on the base.

jrainville committed:
feat(pairing): add config to let users enable message syncing

Fixes https://github.com/status-im/status-desktop/issues/18894

Adds a param for users to enable message syncing during local pairing.

Updates the test to use that param

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (19)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/1/) | 2025-09-24 18:11:19 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/1//consoleText) |
| :heavy_check_mark: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6950/1/) | 2025-09-24 18:11:20 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-180744-c28906-pr6950.zip) |
| :heavy_check_mark: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6950/1/) | 2025-09-24 18:12:49 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-180748-c28906-pr6950.zip) |
| :heavy_check_mark: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6950/1/) | 2025-09-24 18:13:19 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-180753-c28906-pr6950.zip) |
| :heavy_multiplication_x: | c2890689 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/2/) | 2025-09-24 18:16:02 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/2//consoleText) |
| :heavy_check_mark: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/1/) | 2025-09-24 18:17:58 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/1//consoleText) |
| :heavy_check_mark: | c2890689 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6950/1/) | 2025-09-24 18:19:01 | ~11 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250924-180744-c28906-pr6950.zip) |
| | | | | | | |
| :heavy_check_mark: | 7f9287e2 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6950/2/) | 2025-09-24 20:06:37 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-200311-7f9287-pr6950.zip) |
| :heavy_check_mark: | 7f9287e2 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6950/2/) | 2025-09-24 20:07:13 | ~4 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-200307-7f9287-pr6950.zip) |
| :heavy_check_mark: | 7f9287e2 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6950/2/) | 2025-09-24 20:08:39 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250924-200319-7f9287-pr6950.zip) |
| :heavy_check_mark: | 7f9287e2 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/2/) | 2025-09-24 20:12:38 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/2//consoleText) |
| :heavy_check_mark: | 7f9287e2 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6950/2/) | 2025-09-24 20:13:16 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250924-200309-7f9287-pr6950.zip) |
| :heavy_check_mark: | 7f9287e2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/3/) | 2025-09-24 20:29:20 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c4eab7bb | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6950/3/) | 2025-09-26 17:41:15 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250926-173749-c4eab7-pr6950.zip) |
| :heavy_check_mark: | c4eab7bb | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6950/3/) | 2025-09-26 17:43:33 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250926-173754-c4eab7-pr6950.zip) |
| :heavy_check_mark: | c4eab7bb | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6950/3/) | 2025-09-26 17:47:18 | ~9 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250926-173745-c4eab7-pr6950.zip) |
| :heavy_check_mark: | c4eab7bb | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6950/3/) | 2025-09-26 17:55:28 | ~17 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250926-173746-c4eab7-pr6950.zip) |
| :heavy_multiplication_x: | c4eab7bb | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/3/) | 2025-09-26 17:56:58 | ~19 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/3//consoleText) |
| :heavy_check_mark: | c4eab7bb | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/4/) | 2025-09-26 18:13:18 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/4//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | eff97d97 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6950/4/) | 2025-09-29 19:51:14 | ~4 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194621-eff97d-pr6950.zip) |
| :heavy_check_mark: | eff97d97 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6950/4/) | 2025-09-29 19:52:05 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194627-eff97d-pr6950.zip) |
| :heavy_check_mark: | eff97d97 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6950/4/) | 2025-09-29 19:58:54 | ~12 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-250929-194617-eff97d-pr6950.zip) |
| :heavy_check_mark: | eff97d97 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6950/4/) | 2025-09-29 20:09:17 | ~22 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-250929-194618-eff97d-pr6950.zip) |
| :heavy_check_mark: | eff97d97 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/5/) | 2025-09-29 20:18:50 | ~28 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/5//consoleText) |
| :heavy_check_mark: | eff97d97 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/5/) | 2025-09-29 20:26:38 | ~40 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c428c245 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6950/5/) | 2025-10-01 13:43:08 | ~3 min | `linux/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-133942-c428c2-pr6950.zip) |
| :heavy_check_mark: | c428c245 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6950/5/) | 2025-10-01 13:43:15 | ~3 min | `macos/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-133947-c428c2-pr6950.zip) |
| :heavy_check_mark: | c428c245 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6950/5/) | 2025-10-01 13:45:20 | ~5 min | `windows/status-go` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-251001-133955-c428c2-pr6950.zip) |
| :heavy_check_mark: | c428c245 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/6/) | 2025-10-01 13:49:44 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6950/6//consoleText) |
| :heavy_check_mark: | c428c245 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6950/5/) | 2025-10-01 13:50:28 | ~10 min | `linux/nwaku` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/nwaku-251001-133944-c428c2-pr6950.zip) |
| :heavy_check_mark: | c428c245 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/6/) | 2025-10-01 14:06:27 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6950/6//consoleText) |

jrainville committed:
fix(pairing): fix pairing messages using wrong chat ID for 1-1

There was an issue where 1-1 chat messages didn't seem to be synced and it was because we were using the ChatID instead of the LocalChatID, meaning that messages from other people had our own chat ID instead of having their chat ID.

Also upgrades the integration and functional test to validate the behaviour.

jrainville committed:
fix(backup): add missing clock on backup messages

igor-sirotin commented on server/pairing/statecontrol/state_management.go:
```diff
@@ -19,6 +19,9 @@ type ProcessStateManager struct {
 	pairing     bool
 	pairingLock sync.Mutex
 
+	messageSyncEnabled bool
+	messageSyncLock    sync.Mutex
```
Not sure why we need any syncing here.
But `atomic.Bool` is a better option anyway

igor-sirotin commented on tests-functional/tests/test_local_pairing.py:
```diff
@@ -219,17 +219,14 @@ def test_pairing_server_as_sender(self):
-        # assert message_found, "Message sent before pairing not found on paired device"
+        messages = bob_second_device.wakuext_service.chat_messages(sender_chat_id, limit=10)["messages"]
+        assert messages is not None, "No messages found on paired device"
+        message_found = False
+        for message in messages:
+            if message["id"] == message_id and message["text"] == "test_message":
+                message_found = True
+                break
```
some python shenanigans could be applied 😄 
```suggestion
        message_found = any(msg["id"] == message_id and msg["text"] == "test_message" for msg in messages)
```

igor-sirotin commented on tests-functional/tests/test_local_pairing.py:
```diff
@@ -244,7 +241,12 @@ def test_pairing_server_as_receiver(self):
+        # Send a message to Alice before local pairing
+        response = bob.wakuext_service.send_one_to_one_message(alice.public_key, "test_message")
+        message = self.get_message_by_content_type(response, content_type=MessageContentType.TEXT_PLAIN.value)[0]
+        _, sender_chat_id = message["id"], message["chatId"]
```
why get `message["id"]` if we're not using it 🤔

jrainville replied:
It's called tactical copy pasting :upside_down_face:

jrainville commented:
@osmaczko friendly reminder to review

jrainville committed:
test(backup): improve tests by using a map instead of a loop
